### PR TITLE
Kokkos Kernels: updating maintainers and releases

### PR DESCRIPTION
**Pull request by lucbv**
@srajama1 @brian-kelley
Adding the latests releases of Kokkos Kernels and the official maintainers for the package.

### CODE REVIEW

**Review by brian-kelley — APPROVED** [review]

**Comment by lucbv** [comment]
@tldahlgren here is an update of Kokkos Kernels which adds releases that had not been declared previously and a set of maintainers for the package.
The work on the smoke test will be done in a subsequent PR : )

File: var/spack/repos/builtin/packages/kokkos-kernels/package.py [hunk 1]
```diff
@@ -11,12 +11,21 @@ class KokkosKernels(CMakePackage, CudaPackage):
 
     homepage = "https://github.com/kokkos/kokkos-kernels"
     git = "https://github.com/kokkos/kokkos-kernels.git"
-    url = "https://github.com/kokkos/kokkos-kernels/archive/3.1.00.tar.gz"
+    url = "https://github.com/kokkos/kokkos-kernels/archive/3.4.01.tar.gz"
 
     tags = ['e4s']
 
+    test_requires_compiler = True
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/kokkos-kernels/package.py:18]: Did you mean to include this at this point?  The property is intended for smoke tests.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/kokkos-kernels/package.py:18]: It doesn't do any harm to have it here though.

**Review by tldahlgren — APPROVED** [review]
Confirmed the sha256 for all of the versions.

**Comment by tldahlgren** [comment]
I'm going to merge as-is assuming the smoke tests will be coming (soon?) in a pending PR. :grin:

**Comment by lucbv** [comment]
Thanks, I am working on the second PR, I got excited when I had my SSL certificates working correctly so I figured I would push this to test my setup.

**Pull request merged**